cmd/helm/root.go

```diff
@@ -430,5 +433,77 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 	// Find and add plugins
 	loadPlugins(cmd, out)
 
+	// Add template functions which will be used in the usage template
+	cobra.AddTemplateFunc("LoggingFlags", LoggingFlags)
+	cobra.AddTemplateFunc("CapitalizeFlagDescriptions", CapitalizeFlagDescriptions)
+	cmd.SetUsageTemplate(CmdUsageTemplate)
 	return cmd
 }
+
+// LoggingFlags will go through all the logging flags which have the annotation: "log":"flag" and
+// get the appropriate usage
+func LoggingFlags(cmd *cobra.Command) string {
+
+	globalFlags := cmd.Flags()
+	loggingFlags := &pflag.FlagSet{}
+
+	// Retrieve all the flags that have that particular annotation
+	globalFlags.VisitAll(func(f *pflag.Flag) {
+		if reflect.DeepEqual(f.Annotations["log"], []string{"flag"}) {
+			loggingFlags.AddFlag(f)
+		}
+	})
+
+	// Weird hack with cobra13. Essentially, cobra doesn't have a function to return a non-pointer-set of flags
+	// so we'll set the flags as non-hidden, get the descriptions, then set them back to hidden
+	loggingFlags.VisitAll(func(f *pflag.Flag) {
+		f.Hidden = false
+	})
+	usage := CapitalizeFlagDescriptions(loggingFlags)
+	loggingFlags.VisitAll(func(f *pflag.Flag) {
+		f.Hidden = true
+	})
+
+	return usage
+}
+
+// CapitalizeFlagDescriptions adds capitalization to each flag
+func CapitalizeFlagDescriptions(f *pflag.FlagSet) string {
```

Comment:
This may be a difference in opinion, but capitalizing the output of all help output prevents cases where users intend the first word of the output to be intentionally lowercased. There may be cases where the implementor wanted the help output to start with a proper noun (like `helm`, when referencing the CLI). I don't think this is helpful.

If there are certain flags that should be capitalized at the beginning, then let's change those flag docstrings on a case-by-case basis, but I'd argue this should not be enforced globally.

Perhaps some documentation in the [developer guide](https://helm.sh/docs/community/developers/) would be more helpful here. That way developers can make that choice for themselves.

Reply:
If you want to do this for the `klog` flags, then perhaps this can be done on line 462, before marking the flag as hidden.